nio_command_block.py

```diff
@@ -0,0 +1,173 @@
+        title="Reauthenticate Interval",
+        visible=False,
+        default={'seconds': 2400})  # Default to 40 mins
+
+    def __init__(self):
+        super().__init__()
+        self._access_token = None
+        self._reauth_job = None
+
+    def configure(self, context):
+        super().configure(context)
+        if self.security_method == SecurityMethod.OAUTH:
+            self._init_access_token()
+
+    def process_signals(self, signals):
+        for signal in signals:
+            try:
+                url, headers = self._get_url(signal)
+                resp = requests.get(url, headers=headers)
+                sigs = self._process_response(resp)
+                if sigs:
+                    self.notify_signals(sigs)
+            except Exception as e:
+                self._logger.exception(e)
+
+    def _process_response(self, resp):
+        status = resp.status_code
+        if status != 200:
+            self._logger.error("Status {0} returned while requesting : {1}"
+                               .format(status, resp))
+        try:
+            data = resp.json()
+        except:
+            data = resp.text
+        sigs = self._build_signals(data)
+        return sigs
+
+    def _build_signals(self, data):
+        sigs = []
+        if isinstance(data, dict):
+            sigs.append(Signal(data))
+        elif isinstance(data, list):
+            for d in data:
+                sigs.extend(self._build_signals(d))
+        else:
+            sigs.append({'resp': data})
+        return sigs
+
+    def _init_access_token(self):
+        try:
+            self._access_token = self.get_access_token('openid email')
+            self._logger.debug("Obtained access token: {}".format(
+                self._access_token))
+
+            if self._reauth_job:
+                self._reauth_job.cancel()
+
+            # Remember to reauthenticate at a certain point if it's configured
+            if self.reauth_interval.total_seconds() > 0:
+                self._reauth_job = Job(
+                    self._init_access_token, self.reauth_interval, False)
+
+        except OAuth2Exception as oae:
+            self._logger.error(
+                "Error obtaining access token: {}".format(oae))
+            self._access_token = None
+
+    def _get_params(self, signal):
+        """ Return a dictionary of any configured URL parameters """
+        params = dict()
+        for param in self.params:
+            try:
+                params[param.prop_name(signal)] = param.prop_value(signal)
+            except Exception as e:
+                self._logger.error(
+                    'Failed to evaluate command params'.format(e))
+        return params
+
+    def _get_url(self, signal):
+        try:
+            service = self.service_name(signal)
+            block = self.block_name(signal)
+            command = self.command_name(signal)
+        except Exception as e:
+            self._logger.warning(
```

Comment:
Unless this is an error (which I think it might be?), then just use `logger.exception`